### DISCUSSION TITLE
Testing different profiles with varying build-time config?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#24700] [Q&A]
According to https://quarkus.io/guides/getting-started-testing#testing_different_profiles I can create different profiles which have different configuration and then tell different test classes which profile should be used.

That seems like it only works for config which is **NOT** fixed at build time, correct? Is there a way I can do this but have it read different values that are only read at build time?

Basically I have a single test class that has some tests and I want to re-run that test class against different profiles with different build-time config.

**Comment by radcortez** [comment]
If I remember correctly, Quarkus is restarted when a different profile is detected in the execution path. Still, the runtime Config bytecode generator (which stores the build time configuration) is not regenerated.

Can you give me a few examples of build time config that you want to change and test?

**Reply by edeandrea** [reply]
To be honest I don't remember what I was working on when I wrote this :)

I was able to do what I think I was asking about. I created a test profile (https://github.com/datastaxdevs/quarkus-astra-intro-demo/blob/main/src/test/java/com/datastaxdev/todo/DaoTestProfile.java) and then used it (https://github.com/datastaxdevs/quarkus-astra-intro-demo/blob/main/src/test/java/com/datastaxdev/todo/service/MapperAstraServiceTests.java).

I use the profile at build time here (https://github.com/datastaxdevs/quarkus-astra-intro-demo/blob/main/src/main/java/com/datastaxdev/todo/config/AstraConfig.java).

**Reply by ia3andy** [reply]
It's funny to see you here @edeandrea as I am trying to figure out the bug you reported on Quinoa :)

@radcortez I am having the same problem with Quinoa test profile (applied on unit test):
```
public static class EnableAndRunTests implements QuarkusTestProfile {
        @Override
        public Map<String, String> getConfigOverrides() {
            return Map.of(
                    "quarkus.quinoa", "true",
                    "quarkus.quinoa.run-tests", "true");
        }
    }
```
 I get this warning:
```
2023-03-14 08:56:19,877 WARN  [io.qua.run.con.ConfigRecorder] (Quarkus Main Thread) Build time property cannot be changed at runtime:
 - quarkus.quinoa.run-tests is set to 'true' but it is build time fixed to 'false'. Did you change the property quarkus.quinoa.run-tests after building the application?
```

Any thought?

**Reply by radcortez** [reply]
Currently, the test config overrides only applies to the runtime configuration. We don't have a way to override build-time configuration on a test-by-test case.

Probably, adding multiple surefire executions with different build configurations would work.

We can check how easy / hard would be to support this with the test overrides.
